src/containers/Patient.js

```diff
@@ -0,0 +1,62 @@
+import React, { Component, PropTypes } from 'react';
+import { connect } from 'react-redux';
+import { bindActionCreators } from 'redux';
+
+import patientProps from '../prop-types/patient';
+import { fetchPatient, setSelectedPatient} from '../actions/patient';
+import PageHeader from '../components/Header/PageHeader';
+import PatientView from '../components/PatientView/PatientView';
+
+
+class Patient extends Component {
+
+	componentWillMount(){
+		
+		this.props.setSelectedPatient(this.props.params.patient_id);
+		if(!(this.props.params.patient_id == this.props.selectedPatient.id)) {
+			// Fire action to set the selectedPatient correctly
+			this.props.fetchPatient(this.props.params.patient_id);
+		}
+	}
```

Comment:
This seems confusing to me. You select a patient (shouldn't this be done in `PatientListResultsItem`?) and then check to see if the patient you selected is the not the patient you selected (huh?), and if not you fetch the patient, which simply selects the patient with the correct params. Am I missing something?

We have two scenarios:

**1. A patient is selected by firing the `selectPatient` action.**
In this case, the state updates the `selectedPatient` and redirects to the `PatientView` component which should be able to pull all the data from the `selectedPatient` state.

**2. A direct URL link to `Patient/:patient_id` is clicked.**
In this case, we have to first select the patient, then redirect to the `PatientView` component. We would call `fetchPatients` to select the patient.

Perhaps we need to do something like this:
```
componentWillMount(){
  // handle case where directly linked to patient view
  if(!(this.props.params.patient_id == this.props.selectedPatient.id)) {
    this.props.fetchPatient(this.props.params.patient_id);
  }
}
```

Since this only needs to be called when a user is directly linked to the patient view page, we only need to check if the `patient_id` param is not already the `selectedPatient` and only then fetch it if it is not. Otherwise the patient is already selected, correct?

Also, it's tough to test the `selectPatient` and `fetchPatient` actions/reducers because this code doesn't include the patient view yet, so can you explain the difference between them? It looks like `selectPatient` will set `selectedPatient` to be a patient object, whereas `fetchPatient` will set `selectedPatient` to be a patient, encounter, condition, medicationStatement, and riskAssessment object. I'm assuming the latter is what we need so what is the purpose of the former?